src/languages/smtlib2/poly/lexer.mll

```diff
@@ -134,10 +134,19 @@
       | '\n' -> newline lexbuf
       | _ -> ()
     done;
-    (* Check whetehr the symbol is a reserved word. *)
+    (* Check whether the symbol is a reserved word. *)
     try M.find s reserved_words
     with Not_found -> SYMBOL s
 
+  let quoted_symbol newline lexbuf s =
+    (* register the newlines in quoted symbols to maintain correct locations.*)
```

Comment:
But of course (:

Reply:
of course, ^^
note that the newline accounting does two things: 1) update the lexer internal counters for positions and 2) update the `dolmen`-specific table of line starts for a file (which is used to store compact locations, and later recover line counts if needed)